lisa/tools/pkgconfig.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import re
+from typing import Optional, Type, Union
+
+import semver
+from assertpy import assert_that, fail
+from semver import VersionInfo
+
+from lisa.executable import Tool
+from lisa.operating_system import Debian, Fedora, Posix, Suse
+
+coerce_version_info_regex = re.compile(
+    r"v?(?P<major>[0-9]+)\.(?P<minor>[0-9]+)(\.(?P<patch>[0-9]+))?"
+)
+
+
+class Pkgconfig(Tool):
+    @property
+    def command(self) -> str:
+        return "pkg-config"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def install(self) -> bool:
+        assert isinstance(self.node.os, Posix)
+        self.node.os.install_packages("pkg-config")
+        return True
+
+    def _coerce_version_info_string(self, version_string: str) -> VersionInfo:
+        if VersionInfo.isvalid(version_string):
+            return VersionInfo.parse(version_string)
+        # else, attempt to coerce
+        matches = coerce_version_info_regex.search(version_string)
+        assert (
+            matches
+        ), f"Could not coalesce version string {version_string} into a package version"
+
+        major = matches.group("major")
+        minor = matches.group("minor")
+        patch = matches.group("patch")
+
+        assert_that(major).described_as(
+            f"Could not identify major version in version string {version_string}"
+        ).is_not_none()
+        assert_that(minor).described_as(
+            f"Could not identify minor version in version string {version_string}"
+        ).is_not_none()
+
+        coerced_version = ".".join([str(int(x)) for x in [major, minor, patch]])
+        return VersionInfo.parse(coerced_version)
+
+    def package_info_exists(self, package_name: str) -> bool:
+        return self.run(f"--modversion {package_name}").exit_code == 0
+
+    def get_package_version(self, package_name: str) -> VersionInfo:
+        assert_that(self.package_info_exists(package_name)).described_as(
```

Comment:
Did you meet this error before? If there is any error on detection or installation, it should be fix there.

Reply:
Nope, just added it for defensive programming.

Reply:
This step slow down the progress. So it should be removed.

Reply:
Wouldn't it be cached after the first call?

Reply:
I can re-work to make this a little cleaner.

Reply:
re-worked! Made it clearer that there should be one (cacheable) call to run()

Reply:
Oh wait, no that's not right. Don't merge, need more time to fix this.

Reply:
Apologies, I started to second guess what was there. Currently package information *should* always be available for libdpdk when this tool is called.

There is an edge case where you check pkg-config --modversion, see DPDK is not a high enough version, update the package, then check again. In that case you would want to force_run to update the cached package information after the update. This could occur when environments get re-used depending on the test case, specifically in Ubuntu 18.04.

For now I would like to leave that assertion in place; this is new code and we support (and use) many methods for installing DPDK.

Reply:
Since I moved some things around, I'm running additional testing now. I'll update when it's ready for another review.

Reply:
Should good to do, I have additional changes blocked on this so we can go ahead and merge.